### PR TITLE
Feature/bar chart

### LINKED ISSUE #148: Bar Chart
## Description
Create a new component for showing a vertical bar chart in our design system.

## Design
barchart1
barchart2

## Implementation

- Bar chart must take data for a bar that will demonstrate X and Y axis.
- Each bar will take the light gray color based on the design as default
- Each bar may take a color from the user
- On Point hover a tooltip must be shown with information regarding that point if provided

data example
```
// data
[
  { name: 'orfium', subcategory: 'direct', value: 620, barValue: '56%', options: { color: 'the color of branded1', hoverInfo: [ { name: 'Daily Views', value: 450, percentage: '45%' } ] } }
]
```

### PR DESCRIPTION
**Pull request by SVonKarajann**
## Description

Implement a BarChart component for the DS

[Ticket](https://orfium.atlassian.net/secure/RapidBoard.jspa?rapidView=8&projectKey=VID&modal=detail&selectedIssue=VID-151)

[Designs (MVP)](https://app.abstract.com/projects/27ff87f3-129d-472d-a51e-6d9ce0fa3ef0/branches/3f537df7-b688-4d69-a6fc-6247cabf03ae/collections/ed2c27f5-d34d-4f39-b70f-7a59b8d56ac1)

Closes #148 

## Screenshot

barchartformatter
barchartformatter2

### CODE REVIEW

**Comment by elektracodes** [comment]
@SVonKarajann It's Sunday. Go to sleep already

**Comment by SVonKarajann** [comment]
@panvourtsis check if the way i calculate the maxDomainValye and the tickCount fot the xAxis is ok or not 
barchart3

File: src/components/Chart/BarChart/BarChart.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    return data.map(obj => {
+      if (obj?.options?.color) {
+        return obj?.options?.color;
+      }
+
+      return theme.palette.flat.darkBlue[100];
+    });
+  }, [data, theme.palette.flat.darkBlue]);
+
+  const maxLabelLength = useMemo(() => max(data.map(obj => obj.name.length)), [data]);
```
Comment (michalisKout) [inline src/components/Chart/BarChart/BarChart.tsx:82]: you can refactor and also reuse it to line 86 by using the maxBy lodash method:

example:
```
----  utils/common.js

  const findMaxInData = useCallback(operator => () => maxBy(data, operator), [data]);

  const maxLabelLength = findMaxInData(item => item.name.length);
  const maxValue = findMaxInData(obj => obj.value);

  ```

File: src/components/Chart/BarChart/BarChart.tsx [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+  const yAxisWidth = maxLabelLength ? maxLabelLength * multiplyFactor : 60;
+
+  const maxValue = useMemo(() => max(data.map(obj => obj.value)), [data]);
+
+  const { maxDomainValue, tickCount } = getValues(maxValue);
+
+  const labelColoringOptions = useMemo(() => {
```
Comment (michalisKout) [inline src/components/Chart/BarChart/BarChart.tsx:90]: try to refactor this method and the method in line 101 (tickColoringOptions) as they are identical.

File: src/components/Chart/BarChart/components/CustomLabel/CustomLabel.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { LabelProps } from 'recharts';
+import useTheme from 'hooks/useTheme';
+
+interface CustomLabelProps extends LabelProps {
+  colors: Record<string, string>;
+}
+
+const CustomLabel: React.FC<CustomLabelProps> = ({ colors, value, x, y, width, height }) => {
+  const theme = useTheme();
+  const xValue = x && width ? x + width + 16 : 16;
```
Comment (michalisKout) [inline src/components/Chart/BarChart/components/CustomLabel/CustomLabel.tsx:11]: Try to give a name in "magic numbers" (ex. for number 16, 2, and 0) to improve readability  😊✌️.

File: src/components/Chart/BarChart/components/CustomTooltip/CustomTooltip.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+const CustomTooltip: React.FC<TooltipProps> = ({ payload }) => {
+  const options = payload && (payload[0]?.payload?.options?.hoverInfo as HoverInfo[]);
+
+  return options ? (
+    <div className="custom-tooltip" css={tooltipStyle()}>
+      <ul css={tooltipUlStyle()}>
+        {options.map(({ name, value, percentage }) => (
+          <li key={`${name}${value}`} css={tooltipLiStyle()}>
+            <div>{name}</div>{' '}
```
Comment (michalisKout) [inline src/components/Chart/BarChart/components/CustomTooltip/CustomTooltip.tsx:16]: you can remove {' '} in lines 16 and 23.

File: src/components/Chart/BarChart/utils.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  return {
+    maxDomainValue,
+    tickCount,
+  };
+};
+
+export const customTickFormatter = (tick: number, maxDomainValue: number): string => {
+  if (tick === 0) {
+    return `${tick}`;
+  } else if (maxDomainValue > 1000000 || tick === 1000000) {
```
Comment (michalisKout) [inline src/components/Chart/BarChart/utils.ts:25]: You can also give some descriptive names to these magic numbers (1000000 and 1000).

**Review by michalisKout — APPROVED** [review]
LGTM👌

**Comment by SVonKarajann** [comment]
more-merge

**Comment by SVonKarajann** [comment]
Just a reminder @Christos-Zacharopoulos

**Review by Christos-Zacharopoulos — APPROVED** [review]

**Comment by Christos-Zacharopoulos** [comment]
download

**Pull request merged**